Cargo.toml

```diff
@@ -14,3 +14,10 @@ sourmash = "0.15.1"
 anyhow = "1.0.89"
 log = "0.4.22"
 env_logger = "0.11.5"
+
+# For JSON serialization/deserialization
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+
+# For Gzip compression/decompression
+flate2 = "1.0"
```

Comment:
suggest niffler, which allows sniffing/auto-determination of file formats. but we can backport that in if we need.

(I'm using it over in #10)